test/integration/operator_test.go

```diff
@@ -0,0 +1,227 @@
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.RbacV1().ClusterRoles().Get(context.Background(), a13eName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.RbacV1().RoleBindings(namespace).Get(context.Background(), a13eName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.RbacV1().Roles(namespace).Get(context.Background(), sinkName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.RbacV1().RoleBindings(namespace).Get(context.Background(), sinkName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		return nil
+	}, retry.Attempts(10), retry.MaxDelay(3*time.Second))
+
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func checkResourcesAfterDelete(t testing.TB, namespace string, testName Test) {
+	t.Helper()
+
+	clientset, dynaclient, _ := test.GetKubernetesClient()
+
+	err := retry.Do(func() error {
+		gvr := schema.GroupVersionResource{Group: "sources.knative.dev", Version: "v1", Resource: "apiserversources"}
+		_, getErr := dynaclient.Resource(gvr).Namespace(test.K9eNamespace).Get(context.Background(), a13eName, metav1.GetOptions{})
+		if testName == emptyKAC || testName == nonEmptyKAC {
+			if !errs.IsNotFound(getErr) {
+				return errors.New("Unexpectedly found an ApiServerSource.")
+			}
+		} else {
+			if getErr != nil {
+				return getErr
+			}
+		}
+		_, getErr = clientset.CoreV1().ConfigMaps(test.K9eNamespace).Get(context.Background(), filtersName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.CoreV1().ServiceAccounts(test.K9eNamespace).Get(context.Background(), a13eName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.RbacV1().ClusterRoles().Get(context.Background(), a13eName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.RbacV1().RoleBindings(namespace).Get(context.Background(), a13eName, metav1.GetOptions{})
+		if !errs.IsNotFound(getErr) {
+			return errors.New("Unexpectedly found Rolebinding " + a13eName + " in namespace " + namespace + ".")
+		}
+		_, getErr = clientset.RbacV1().Roles(namespace).Get(context.Background(), sinkName, metav1.GetOptions{})
+		if !errs.IsNotFound(getErr) {
+			return errors.New("Unexpectedly found Role " + sinkName + " in namespace " + namespace + ".")
+		}
+		_, getErr = clientset.RbacV1().RoleBindings(namespace).Get(context.Background(), sinkName, metav1.GetOptions{})
+		if !errs.IsNotFound(getErr) {
+			return errors.New("Unexpectedly found Rolebinding " + sinkName + " in namespace " + namespace + ".")
+		}
+		return nil
+	}, retry.Attempts(10), retry.MaxDelay(3*time.Second))
+
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func createKAC(namespace string, resources map[string]any) *unstructured.Unstructured {
```

Comment:
this name is a little missleading, at first I thought that the `createKAC` function was taking care of creating the resource in the cluster, but its only instantiating it. How about `newKAC`? 

```suggestion
func newKAC(namespace string, resources map[string]any) *unstructured.Unstructured {
```

Reply:
Fixed.